mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -713,6 +713,15 @@ private[spark] object RandomForest extends Logging {
         binAggregates.metadata.numSplits(featureIndex) != 0
       }
 
+    // If no valid splits for features, then this split is invalid,
+    // return invalid information gain stats.  Take any split and continue.
+    if (validFeatureSplits.isEmpty) {
+      val parentImpurityCalculator = binAggregates.getImpurityCalculator(0, 0)
+      // Splits is empty, so arbitrarily choose to split on first feature with any threshold
+      return (new ContinuousSplit(0, 0),
+        ImpurityStats.getInvalidImpurityStats(parentImpurityCalculator))
+    }
+
     // For each (feature, split), calculate the gain, and select the best (feature, split).
     val (bestSplit, bestSplitStats) =
```

Comment:
I slightly prefer to avoid using `if (...) return` and instead do:

````scala
val splitsAndImpurityInfo = validFeatureSplits.map { ... }
val (bestSplit, bestSplitStats) = if (splitsAndImpurityInfo.isEmpty) { ... } else { ... }
(bestSplit, bestSplitStats)
````